src/cli/lint/autofix.test.ts

```diff
@@ -0,0 +1,236 @@
+  return `\n${result}`;
+};
+
+const expectAutofixCommit = () => {
+  expect(runESLint).toHaveBeenCalledTimes(1);
+  expect(runPrettier).toHaveBeenCalledTimes(1);
+  expect(Git.commitAllChanges).toHaveBeenCalledTimes(1);
+};
+
+const expectNoAutofix = () => {
+  expect(runESLint).not.toHaveBeenCalled();
+  expect(runPrettier).not.toHaveBeenCalled();
+  expect(Git.commitAllChanges).not.toHaveBeenCalled();
+  expect(Git.push).not.toHaveBeenCalled();
+};
+
+beforeEach(() => {
+  delete process.env.BUILDKITE_PIPELINE_DEFAULT_BRANCH;
+  delete process.env.GITHUB_ACTIONS;
+  delete process.env.GITHUB_REF_PROTECTED;
+
+  process.env.CI = 'true';
+
+  jest
+    .spyOn(console, 'log')
+    .mockImplementation((...args) => stdoutMock(`${args.join(' ')}\n`));
+});
+
+afterEach(jest.resetAllMocks);
+
+describe('autofix', () => {
+  it('bails on a non-CI environment', async () => {
+    delete process.env.CI;
+
+    await expect(autofix({ debug: false })).resolves.toBeUndefined();
+
+    expectNoAutofix();
+  });
+
+  it('bails on the master branch', async () => {
+    jest.mocked(Git.currentBranch).mockResolvedValue('master');
+
+    await expect(autofix({ debug: false })).resolves.toBeUndefined();
+
+    expectNoAutofix();
+  });
+
+  it('bails on the main branch', async () => {
+    jest.mocked(Git.currentBranch).mockResolvedValue('main');
+
+    await expect(autofix({ debug: false })).resolves.toBeUndefined();
+
+    expectNoAutofix();
+  });
+
+  it('bails on the Buildkite default branch', async () => {
+    process.env.BUILDKITE_PIPELINE_DEFAULT_BRANCH = 'devel';
+
+    jest.mocked(Git.currentBranch).mockResolvedValue('devel');
+
+    await expect(autofix({ debug: false })).resolves.toBeUndefined();
+
+    expectNoAutofix();
+  });
+
+  it('bails on a GitHub protected branch', async () => {
+    process.env.GITHUB_REF_PROTECTED = 'true';
+
+    jest.mocked(Git.currentBranch).mockResolvedValue('beta');
+
+    await expect(autofix({ debug: false })).resolves.toBeUndefined();
+
+    expectNoAutofix();
+  });
+
+  it('bails on an autofix head commit', async () => {
+    jest.mocked(Git.currentBranch).mockResolvedValue('feature');
+    jest
+      .mocked(Git.getHeadCommitMessage)
+      .mockResolvedValue('Run `skuba format`');
+
+    await expect(autofix({ debug: false })).resolves.toBeUndefined();
+
+    expectNoAutofix();
+  });
+
+  it('skips push on empty commit', async () => {
+    jest.mocked(Git.commitAllChanges).mockResolvedValue(undefined);
+
+    await expect(autofix({ debug: false })).resolves.toBeUndefined();
+
+    expectAutofixCommit();
+    expect(Git.push).not.toHaveBeenCalled();
+
+    expect(stdout()).toMatchInlineSnapshot(`
+      "
+
+      Trying to autofix with ESLint and Prettier...
+      No autofixes detected.
+      "
+    `);
+  });
+
+  it('uses Git CLI in GitHub Actions', async () => {
+    process.env.GITHUB_ACTIONS = 'true';
+
+    const push = jest.fn();
+    jest.mocked(simpleGit).mockReturnValue({ push } as any);
```

Comment:
any? 👀

Reply:
Yeah, unfortunately this would be annoying to mock in its entirety. We disable the `any` linting rules in tests for such scenarios and I'm lazy to DI here 😛